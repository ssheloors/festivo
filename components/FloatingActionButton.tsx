import { styled } from "tamagui";

import { Button } from "./Button";

export const FloatingActionButton = styled(Button, {
  theme: "accent",
  bottom: "$10",
  left: 0,
  right: 0,
  margin: "auto",
  borderRadius: "$12",
});
