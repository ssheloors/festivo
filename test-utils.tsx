import { QueryClientProvider, QueryClient } from "@tanstack/react-query";
import { render, RenderOptions } from "@testing-library/react-native";
import * as expoRouter from "expo-router";
import { ReactElement, ReactNode } from "react";
import { TamaguiProvider } from "tamagui";

import { PayloadClientContext } from "./components/PayloadClientProvider";
import * as useStorageModule from "./hooks/use-storage";
import { createPayloadTestClient } from "./utils/payload-client";

import tamaguiConfig from "@/tamagui.config";

global.console = {
  ...console,
  // uncomment to ignore a specific log level
  // log: jest.fn(),
  // debug: jest.fn(),
  // info: jest.fn(),
  // warn: jest.fn(),
  // error: jest.fn(),
};
global.alert = jest.fn();

let mockedRouterParams: expoRouter.UnknownOutputParams = {};

jest.mock("expo-router");
const mockedRouter = jest.mocked(expoRouter);
mockedRouter.useLocalSearchParams.mockImplementation(() => mockedRouterParams);
mockedRouter.Link = jest.requireActual("expo-router").Link;

jest.mock("@react-navigation/bottom-tabs", () => ({
  ...jest.requireActual("@react-navigation/bottom-tabs"),
  useBottomTabBarHeight: () => 0,
}));

jest.mock("./hooks/use-storage.ts");
jest.mock("./hooks/use-user.ts");

const queryClient = new QueryClient();
export let payloadTestClient = createPayloadTestClient();
beforeEach(() => {
  payloadTestClient = createPayloadTestClient();
});

export function AllTheProviders({ children }: { children: ReactNode }) {
  return (
    <QueryClientProvider client={queryClient}>
      <PayloadClientContext.Provider value={payloadTestClient}>
        <TamaguiProvider config={tamaguiConfig}>{children}</TamaguiProvider>
      </PayloadClientContext.Provider>
    </QueryClientProvider>
  );
}

export function setLocalSearchParams(params: expoRouter.UnknownOutputParams) {
  mockedRouterParams = params;
}
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const memoryStorage = (useStorageModule as any).memoryStorage as Map<
  string,
  string
>;
beforeEach(() => {
  memoryStorage.clear();
});

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: AllTheProviders, ...options });

/* eslint-disable import/export */
export * from "@testing-library/react-native";
export { customRender as render };
/* eslint-enable */
