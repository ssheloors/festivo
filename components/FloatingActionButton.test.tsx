import { FloatingActionButton } from "./FloatingActionButton";

import { render } from "@/test-utils";

it(`renders correctly`, () => {
  const { asFragment } = render(
    <FloatingActionButton>Foo</FloatingActionButton>,
  );

  expect(asFragment()).toMatchSnapshot();
});
