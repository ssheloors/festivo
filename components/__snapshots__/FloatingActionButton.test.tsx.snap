// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <span
    class=""
    style="display: contents;"
  >
    <span
      class=" _dsp_contents"
    >
      <span
        class=" t_dark _dsp_contents is_Theme"
      >
        <span
          class=" _dsp_contents"
        >
          <span
            class="t_sub_theme t_accent_Button _dsp_contents is_Theme"
          >
            <button
              class="is_Button _bg-0hover-backgroundH3423444 _btc-0hover-borderColor69916956 _brc-0hover-borderColor69916956 _bbc-0hover-borderColor69916956 _blc-0hover-borderColor69916956 _bg-0active-backgroundP3496915 _btc-0active-borderColor77378595 _brc-0active-borderColor77378595 _bbc-0active-borderColor77378595 _blc-0active-borderColor77378595 _outlineColor-0focus-visible-outlineColo114 _outlineStyle-0focus-visible-solid _outlineWidth-0focus-visible-2px _dsp-flex _ai-center _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _fd-row _pr-t-space-tru101 _pl-t-space-tru101 _h-t-size-true _btlr-t-radius-12 _btrr-t-radius-12 _bbrr-t-radius-12 _bblr-t-radius-12 _jc-center _fw-nowrap _cur-pointer _bg-background _btw-1px _brw-1px _bbw-1px _blw-1px _btc-transparent _brc-transparent _bbc-transparent _blc-transparent _bbs-solid _bts-solid _bls-solid _brs-solid _b-t-space-10 _l-0px _r-0px _mt-auto _mr-auto _mb-auto _ml-auto  is_Button "
              data-disable-theme="true"
              role="button"
              tabindex="0"
            >
              <span
                class=" _dsp_contents"
              >
                <span
                  class="t_sub_theme t_accent_Button _dsp_contents is_Theme"
                >
                  <span
                    class="is_Button font_body _dsp-inline _bxs-border-box _ww-break-word _whiteSpace-nowrap _mt-0px _mr-0px _mb-0px _ml-0px _ff-f-family _fow-f-weight-4 _ls-f-letterSpa1360334202 _fos-f-size-4 _lh-f-lineHeigh112923 _col-color _ussel-none _cur-pointer _fg-0 _fs-1 _maw-10037 _ox-hidden _oy-hidden _textOverflow-ellipsis"
                  >
                    Foo
                  </span>
                </span>
              </span>
            </button>
          </span>
        </span>
        <div
          style="display: contents;"
        />
      </span>
    </span>
  </span>
</DocumentFragment>
`;
